### PR TITLE
Major performance improvements on browsing rows

### PR DESCRIPTION
**Pull request by williamdes**
I made some major improvements to the performance that browsing rows had.

Using: xhprof

Ref: #16005

Explained:
- The first change is to just move a variable to avoid it to be re-built
- The second one is splitting in 3 functions the description array to avoid wasting time fetching translations that will not be used (99% are not used, only one is picked up) (yes I built a script to check all the keys and values exist and are the same than before, diff should be okay)

Dataset and method:
- A table of more that 1.1 Gio of data
- 7 973 078 rows
- show rows by 500 on browse page

### Before (14 229ms)
image

#### Call refresh button (14 981ms)
image

### After (7 524 ms)
image

#### Call refresh button (6 673 ms)
image

### CODE REVIEW

File: libraries/classes/Config/Descriptions.php [hunk 1]
```diff
@@ -54,666 +52,337 @@ public static function get($path, $type = 'name')
...
-    /**
-     * Return name or description for a cleaned up configuration path.
-     *
-     * @param string $path Path of configuration
-     * @param string $type Type of message, either 'name', 'cmt' or 'desc'
-     *
-     * @return string|null Null if not found
-     */
-    public static function getString($path, $type = 'name')
+    private static function getCmt(): array
```
Comment (mauriciofauth) [inline libraries/classes/Config/Descriptions.php:55]:
```suggestion
    private static function getComments(): array
```

File: libraries/classes/Config/Descriptions.php [hunk 2]
```diff
@@ -997,6 +995,27 @@ public static function getString($path, $type = 'name')
...
+     * @param string $type Type of message, either 'name', 'cmt' or 'desc'
+     *
+     * @return string|null Null if not found
+     */
+    public static function getString($path, $type = 'name'): ?string
+    {
+        $descriptions = [];
+
+        if ($type === 'cmt') {
+            $descriptions = self::getCmt();
```
Comment (mauriciofauth) [inline libraries/classes/Config/Descriptions.php:1013]:
```suggestion
            $descriptions = self::getComments();
```

**Review by MauricioFauth — APPROVED** [review]
Really nice. :1st_place_medal:

**Comment by ibennetch** [comment]
I'm very impressed with the speed improvements you've made here. Nice work!

**Comment by ibennetch** [comment]
I'd like to get this in 5.1 before I release 5.1.0rc2.

**Comment by williamdes** [comment]
> I'm very impressed with the speed improvements you've made here. Nice work!

Thank you @ibennetch !

> Really nice. 1st_place_medal:

Thank you @mauriciofauth !

> I'd like to get this in 5.1 before I release 5.1.0rc2.

I will merge that right now, could you please review #16584 ?


NB: some discussion is going on at https://github.com/code-lts/welcome/issues/6 about xhprof, I intend to make it usable for other people so our users can send us xhprof files

**Pull request merged**